homeassistant/components/climate/myicomfort.py

```diff
@@ -0,0 +1,225 @@
+"""
+Support for the Lennox iComfort WiFi thermostat.
+
+Sample entry for configuration.yaml:
+
+climate:
+  - platform: myicomfort
+    name: Downstairs
+    username: YOUR_MYICOMFORT_USERNAME
+    password: YOUR_MYICOMFORY_PASSWORD
+
+"""
+
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.climate import (
+    ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW,
+    ClimateDevice, PLATFORM_SCHEMA, STATE_AUTO,
+    STATE_COOL, STATE_HEAT, SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_TARGET_TEMPERATURE_HIGH, SUPPORT_TARGET_TEMPERATURE_LOW,
+    SUPPORT_OPERATION_MODE, SUPPORT_AWAY_MODE, SUPPORT_FAN_MODE)
+from homeassistant.const import (
+    CONF_USERNAME, CONF_PASSWORD, TEMP_CELSIUS, TEMP_FAHRENHEIT,
+    STATE_ON, STATE_OFF, STATE_IDLE, ATTR_TEMPERATURE)
+
+REQUIREMENTS = ['myicomfort==0.2.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+# HA doesn't have a 'circulate' state defined.
+STATE_CIRCULATE = 'circulate'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_TARGET_TEMPERATURE_HIGH |
+                 SUPPORT_TARGET_TEMPERATURE_LOW |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_AWAY_MODE |
+                 SUPPORT_FAN_MODE)
+
+FAN_MODES = [
+    STATE_AUTO, STATE_ON, STATE_CIRCULATE
+]
+
+OP_MODES = [
+    STATE_OFF, STATE_HEAT, STATE_COOL, STATE_AUTO
+]
+
+SYSTEM_STATES = [
+    STATE_IDLE, STATE_HEAT, STATE_COOL
+]
+
+TEMP_UNITS = [
+    TEMP_FAHRENHEIT, TEMP_CELSIUS
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional('name', default='iComfort'): cv.string,
+    vol.Optional('system', default=0): vol.Coerce(int),
+    vol.Optional('zone', default=0): vol.Coerce(int),
+    vol.Optional('min_temp'): vol.Coerce(float),
+    vol.Optional('max_temp'): vol.Coerce(float),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the climate platform."""
+    from myicomfort.api import Tstat
+
+    username = config.get(CONF_USERNAME)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.